apollo-router/src/context/mod.rs

```diff
@@ -246,6 +303,66 @@ impl Default for Context {
     }
 }
 
+/// Convert context key to the deprecated context key (mainly useful for coprocessor/rhai)
+pub(crate) fn context_key_to_deprecated(key: String) -> String {
+    match key.as_str() {
+        OPERATION_NAME => DEPRECATED_OPERATION_NAME.to_string(),
+        OPERATION_KIND => DEPRECATED_OPERATION_KIND.to_string(),
+        APOLLO_AUTHENTICATION_JWT_CLAIMS => DEPRECATED_APOLLO_AUTHENTICATION_JWT_CLAIMS.to_string(),
+        AUTHENTICATION_REQUIRED_KEY => DEPRECATED_AUTHENTICATION_REQUIRED_KEY.to_string(),
+        REQUIRED_SCOPES_KEY => DEPRECATED_REQUIRED_SCOPES_KEY.to_string(),
+        REQUIRED_POLICIES_KEY => DEPRECATED_REQUIRED_POLICIES_KEY.to_string(),
+        APOLLO_OPERATION_ID => DEPRECATED_APOLLO_OPERATION_ID.to_string(),
+        UNRESOLVED_LABELS_KEY => DEPRECATED_UNRESOLVED_LABELS_KEY.to_string(),
+        LABELS_TO_OVERRIDE_KEY => DEPRECATED_LABELS_TO_OVERRIDE_KEY.to_string(),
+        FIRST_EVENT_CONTEXT_KEY => DEPRECATED_FIRST_EVENT_CONTEXT_KEY.to_string(),
+        CLIENT_NAME => DEPRECATED_CLIENT_NAME.to_string(),
+        CLIENT_VERSION => DEPRECATED_CLIENT_VERSION.to_string(),
+        STUDIO_EXCLUDE => DEPRECATED_STUDIO_EXCLUDE.to_string(),
+        SUBGRAPH_FTV1 => DEPRECATED_SUBGRAPH_FTV1.to_string(),
+        COST_ESTIMATED_KEY => DEPRECATED_COST_ESTIMATED_KEY.to_string(),
+        COST_ACTUAL_KEY => DEPRECATED_COST_ACTUAL_KEY.to_string(),
+        COST_RESULT_KEY => DEPRECATED_COST_RESULT_KEY.to_string(),
+        COST_STRATEGY_KEY => DEPRECATED_COST_STRATEGY_KEY.to_string(),
+        ENABLED_CONTEXT_KEY => DEPRECATED_ENABLED_CONTEXT_KEY.to_string(),
+        FORMATTED_QUERY_PLAN_CONTEXT_KEY => DEPRECATED_FORMATTED_QUERY_PLAN_CONTEXT_KEY.to_string(),
+        QUERY_PLAN_CONTEXT_KEY => DEPRECATED_QUERY_PLAN_CONTEXT_KEY.to_string(),
+        PERSISTED_QUERY_CACHE_HIT => DEPRECATED_PERSISTED_QUERY_CACHE_HIT.to_string(),
+        PERSISTED_QUERY_REGISTERED => DEPRECATED_PERSISTED_QUERY_REGISTERED.to_string(),
+        _ => key,
```

Comment:
I think it would be better to return `Result` or `Option`. Otherwise silent failure is possible. Or make the "keys" members of an `Enum` so it can't be invoked with a bad key.

Reply:
I'm not sure to understand the idea here. Basically it's used directly to convert a deprecated key to a current one. If the key is not listed as deprecated then I just use the original one so it's not an error it's just a conversion and converting the context key to the original itself is not an error IMHO

Reply:
I was thinking it might be possible to provide a key which was a typo and so you'd get yourself back and not notice there was a problem.

Reply:
Hmm got it. I think that's something we should do at the jsonschema level for the selective configuration. This function is only used when `deprecated` is enabled and so you don't need to provide any context key, it's directly coming from our context itself.

Reply:
Ok. It feels like a "dangerous" function. Could you add a comment that captures what you've just written to the conversion functions? That way if anyone tries to use them in future they'll be aware of the constraints.

Reply:
It'd be hard to enforce context keys that actually exist because users can add their own, but maybe we could do it for `apollo::` namespaced ones

Reply:
fixed in https://github.com/apollographql/router/pull/6752/commits/50fa970adefa94c1bc67b8a5c2919f29ec73801e